source/common/local_reply/local_reply.cc

```diff
@@ -37,8 +35,13 @@ class BodyFormatter {
               const Http::ResponseTrailerMap& response_trailers,
               const StreamInfo::StreamInfo& stream_info, std::string& body,
               absl::string_view& content_type) const {
-    body = formatter_->formatWithContext(
-        {&request_headers, &response_headers, &response_trailers, body}, stream_info);
+    // No specific formatter is provided and the default formatter %LOCAL_REPLY_BODY% will
+    // be used. That means the body will be the same as the original body and we don't need
+    // to format it.
+    if (formatter_ != nullptr) {
```

Comment:
was formatter ever null before? should we envoy bug somewhere if it is?

Reply:
In previous implmentation, formatter_ will be never be null. Because the `BodyFormatter` will parse  `%LOCAL_REPLY_BODY%` and get a default formatter. The default `BodyFormatter` will be used by the HCM in the envoy mobile. This is why the envoy mobile will depends on substitution formatter provider.

In this PR, we removed this default formatter (and break the dependency from envoy mobile to the substitution formatter provider) because the `%LOCAL_REPLY_BODY%` here won't update the body and make no sense anyway.

Reply:
What's the default formatter if things are linked in?  If the linked in behavior is different from not I think it's worth a logged warning at least, and envoy bug at most.
Just removed the direct response code from E-M https://github.com/envoyproxy/envoy/pull/35470 so we can go either way

Reply:
The code behavior will not be effected by the linked in behavior. This PR also doesn't change any behavior.

In the previous implementation, the default formatter is formatter that parsed from %LOCAL_REPLY_BODY%, which actually do nothing and make no sense except introduce an additional string copy.
In the new implementation, these is no default formatter and is nullptr.